docs/src/examples.md

```diff
@@ -0,0 +1,609 @@
+
+# In this file, our `date` column has dates that are formatted like `yyyy/mm/dd`. We can pass just such a string to the `dateformat` keyword argument to tell parsing to use it when looking for `Date` or `DateTime` columns. Note that currently, only a single `dateformat` string can be passed to parsing, meaning multiple columns with different date formats cannot all be parsed as `Date`/`DateTime`.
+data = """
+code,date
+0,2019/01/01
+1,2019/01/02
+"""
+
+file = CSV.File(file; dateformat="yyyy/mm/dd")
+```
+
+### [Custom decimal separator](@id decimal_example)
+
+```julia
+using CSV
+
+# In many places in the world, floating point number decimals are separated with a comma instead of a period (`3,14` vs. `3.14`). We can correctly parse these numbers by passing in the `decimal=','` keyword argument. Note that we probably need to explicitly pass `delim=';'` in this case, since the parser will probably think that it detected `','` as the delimiter.
+data = """
+col1;col2;col3
+1,01;2,02;3,03
+4,04;5,05;6,06
+"""
+
+file = CSV.File(file; delim=';', decimal=',')
+```
+
+### [Custom bool strings](@id truestrings_example)
+
+```julia
+using CSV
+
+# By default, parsing only considers the string values `true` and `false` as valid `Bool` values. To consider alternative values, we can pass a `Vector{String}` to the `truestrings` and `falsestrings` keyword arguments.
+data = """
+id,paid,attended
+0,T,TRUE
+1,F,TRUE
+2,T,FALSE
+3,F,FALSE
+"""
+
+file = CSV.File(file; truestrings=["T", "TRUE"], falsestrings=["F", "FALSE"])
+```
+
+### [Matrix-like Data](@id matrix_example)
+
+```julia
+using CSV
+
+# This file contains a 3x3 identity matrix of `Float64`. By default, parsing will detect the delimiter and type, but we can also explicitly pass `delim= ' '` and `types=Float64`, which tells parsing to explicitly treat each column as `Float64`, without having to guess the type on its own.
+data = """
+1.0 0.0 0.0
+0.0 1.0 0.0
+0.0 0.0 1.0
+"""
+
+file = CSV.File(file; header=false)
+file = CSV.File(file; header=false, delim=' ', types=Float64)
+```
+
+### [Providing types](@id types_example)
+
+```julia
+using CSV
+
+# In this file, our 3rd column has an invalid value on the 2nd row `invalid`. Let's imagine we'd still like to treat it as an `Int` column, and ignore the `invalid` value. The syntax examples provide several ways we can tell parsing to treat the 3rd column as `Int`, by referring to column index `3`, or column name with `Symbol` or `String`. We can also provide an entire `Vector` of types for each column (and which needs to match the length of columns in the file). There are two additional keyword arguments that control parsing behavior; in the first 4 syntax examples, we would see a warning printed like `"warning: invalid Int64 value on row 2, column 3"`. In the fifth example, passing `silencewarnings=true` will suppress this warning printing. In the last syntax example, passing `strict=true` will result in an error being thrown during parsing.
```

Comment:
Maybe, a nitpicking suggestion: Place each sentence on a newline to ease review of the Git diff in the future.

Reply:
Good suggestion! Done.